packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -126,4 +129,33 @@ export default class OperatorModeStateService extends Service {
 
     return newState;
   }
+
+  async constructRecentCards() {
+    const recentCardIdsString = localStorage.getItem('recent-cards');
+    if (!recentCardIdsString) {
+      return;
+    }
+
+    const recentCardIds = JSON.parse(recentCardIdsString) as string[];
+    for (const recentCardId of recentCardIds) {
+      const card = await this.cardService.loadModel(new URL(recentCardId));
+      this.recentCards.push(card);
+    }
+  }
+
+  async addRecentCards(card: Card) {
+    const existingCardIndex = this.recentCards.findIndex(
+      (recentCard) => recentCard.id === card.id
+    );
+    if (existingCardIndex !== -1) {
+      this.recentCards.splice(existingCardIndex, 1);
+    }
+
+    this.recentCards.push(card);
+    if (this.recentCards.length > 10) {
+      this.recentCards = new TrackedArray<Card>(this.recentCards.slice(1));
```

Comment:
why are we creating a new TrackedArray here? can't you just mutate the array you already have with a splice like you do above?